### PR TITLE
bug 1746019: Add event recording for machines

### PR DESCRIPTION
**Pull request by iamemilio**
**What this PR does / why we need it**: Enables Event recording so that the interface is comparable to CAPA does things

**Which issue(s) this PR fixes**: https://bugzilla.redhat.com/show_bug.cgi?id=1746019

### CODE REVIEW

File: pkg/cloud/openstack/machine/actuator.go [hunk 1]
```diff
@@ -203,7 +214,7 @@ func (oc *OpenstackClient) Create(ctx context.Context, cluster *clusterv1.Cluste
 	// `cluster` object so it's `nil` here. Read the cluster name from
 	// the `machine` instead.
 	var clusterName string
-	// TODO(egarcia): if we ever use the cluster object, this will benifit from reading from it
+	// TODO(egarcia): if we ever use the cluster object, this wiupdateEventActionll benifit from reading from it
```
Comment (Fedosin) [inline pkg/cloud/openstack/machine/actuator.go]: A small typo here

File: config/rbac/rbac_role.yaml [hunk 2]
```diff
@@ -53,3 +52,12 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
```
Comment (Fedosin) [inline config/rbac/rbac_role.yaml]: do we really use this file during the deployment?

Comment (iamemilio) [reply config/rbac/rbac_role.yaml]: yeah, I was surprised too

Comment (Fedosin) [reply config/rbac/rbac_role.yaml]: I mean that last time I checked there was no cluster role "openstack-provider-manager-role", I need to take a look again

Comment (iamemilio) [reply config/rbac/rbac_role.yaml]: all I know is that it failed until I added this ¯\\_(ツ)_/¯

Comment (iamemilio) [reply config/rbac/rbac_role.yaml]: https://github.com/openshift/machine-api-operator/blob/master/install/0000_30_machine-api-operator_09_rbac.yaml

**Comment by ingvagabund** [comment]
/retitle bug 1746019: Add event recording for machines

**Comment by Fedosin** [comment]
Just checked - all workers have events now. Thanks @iamemilio !

**Pull request merged**